src/pip/_internal/cli/base_command.py

```diff
@@ -87,13 +87,10 @@ def __init__(self, name, summary, isolated=False):
             self.parser,
         )
         self.parser.add_option_group(gen_opts)
-        try:
-            # mypy raises error due to
-            # https://github.com/python/mypy/issues/5868
-            self.add_options(self.cmd_opts)  # type: ignore
-        except AttributeError:
-            # it means that the base class has not defined the method
-            logger.debug("No add_options method defined in the class")
+
+        # mypy raises error due to
+        # https://github.com/python/mypy/issues/5868
+        self.add_options()  # type: ignore
```

Comment:
```suggestion
        self.add_options()
```

Reply:
It was not working because `mypy` was producing errors. That' why I referred to the linked `mypy` issue.

Reply:
And I think, now with latest changes, this won't be an issue.